Tools/AP_Periph/can.cpp

```diff
@@ -2329,6 +2330,82 @@ void AP_Periph_FW::can_rangefinder_update(void)
 }
 
 
+void AP_Periph_FW::can_proximity_update()
+{
+#ifdef HAL_PERIPH_ENABLE_PRX
+    if (proximity.get_type(0) == AP_Proximity::Type::None) {
+        return;
+    }
+#if CAN_PROBE_CONTINUOUS
+    if (proximity.num_sensors() == 0) {
+        uint32_t now = AP_HAL::native_millis();
+        static uint32_t last_probe_ms;
+        if (now - last_probe_ms >= 1000) {
+            last_probe_ms = now;
+            proximity.init();
```

Comment:
how long does init() take? could stall other tasks on the main thread

Reply:
Init takes about 400-500 microseconds on my L431, depending on the sensor selected